thorlcr/graph/thgraph.hpp

```diff
@@ -121,14 +121,22 @@ interface IThorResult : extends IInterface
     virtual const void * getLinkedRowResult() = 0;
 };
 
+enum ThorGraphResultType:unsigned
+{
+    thorgraphresult_nul = 0x00,
+    thorgraphresult_distributed = 0x01,
+    thorgraphresult_grouped = 0x02,
+    thorgraphresult_sparse = 0x04
+};
+inline ThorGraphResultType mergeResultTypes(ThorGraphResultType l, ThorGraphResultType r) { return (ThorGraphResultType) (l|r); }
```

Comment:
I just had an idea.  Whenever we have enums that are composed of bitfields we could have the following:

    inline ThorGraphResultType operator |(ThorGraphResultType l, ThorGraphResultType r) { return (ThorGraphResultType) ((unsigned)l(unsigned)|r); }

So that you can use the | operator.  (I just tried it on my RoxieHeapFlags enumeration, and it seemed to work.)